src/app/ReadHandler.h

```diff
@@ -491,6 +491,9 @@ class ReadHandler : public Messaging::ExchangeDelegate
     /// @param aFlag Flag to clear
     void ClearStateFlag(ReadHandlerFlags aFlag);
 
+    void UseLargePayloadBuffer() { mUseLargePayloadBuffer = true; }
+    bool ShouldUseLargePayloadBuffer() { return mUseLargePayloadBuffer; }
```

Comment:
I wonder if instead of pushing a bool down the stack if maybe ReadHandler should instead have the ability to return correctly sized data instead.

Do we expect to need `ShouldUseLargePayloadBuffer` more than just creating a buffer?
Otherwise it may be that we could split logic a bit more by having `ReadHandler::AllocatePacketForReply` that returns a correctly sized buffer instead and localizes the size decisions.

Reply:
These are now removed. The size inference is being done in-band within ReadHandler and returned by ReadHandler::GetReportBufferMaxSize()